### DISCUSSION TITLE
Experiments as panel data - How to forecast with sktime ?

### DISCUSSION THREAD

**Discussion by formont2u** [alan-turing-institute/sktime#4235] [Q&A] [answered]
Hello everyone, 

I am using sktime to forecast data from different experiment. and I am fairly new to sktime.  My dataset has the structure of a multi-index panda dataframe. I have 10 experiments of different lengths and each experiment comprises a time range t. In total there are 3 exogenous variables and  2 endogenous ones.  

The training set consist sof 8 full experiments. The two other experiments are split the regular way, train and test split. The training set is therefore quite 'original' as it comprises of 8 full experiments of different length+ a fraction of the 2 other experiments. The test set comprises the other fraction of the 2 experiments.

The goal is to forecast the two splits left, one by one.

I have two major issues that I am dealing with : 

1. I have managed to do some forecasts, and even though they are OK, it seems that most of the forecasters I have managed to use are only able to forecast one experiment at a time. Therefore transformer.transformers_ gives me a list of fit forecasters. I wanted to know if there are forecasters that can fit a whole multi-index Dataframe  ?
2. My goal is to build a model that only gives predictions for one experiment.  It results from my first problem, but I am unable to generate predictions for one experiment. I aim to have 8 full experiments in the training set, give the start of my new experiment (end+ex) and generate a forecast for that one experiment only.  

I will keep looking for a solution to this problem and if someone has encountered the same difficulties I would gladly know how you can solve this problem :)

**Comment by fkiraly** [comment, marked as answer]
I think this is the same issue or feature request as in this issue:
https://github.com/sktime/sktime/issues/4209
Could you kindly check whether it looks like the same?

Seems like a common request, I've also run into that - perhaps a good thing to put on the roadmap.

**Reply by formont2u** [reply]
This is definitely a similar issue, thank you for the answer.
